mmedit/datasets/pipelines/matlab_resize.py

```diff
@@ -0,0 +1,270 @@
+from math import ceil
+
+import numpy as np
+
+from ..registry import PIPELINES
+
+
+def get_size_from_scale(input_size, scale_factor):
+    """Get the output size given input size and scale factor.
+
+    Args:
+        input_size (tuple): The size of the input image.
+        scale_factor (float): The resize factor.
+
+    Returns:
+        list[int]: The size of the output image.
+    """
+
+    output_shape = [
+        int(ceil(scale * shape))
+        for (scale, shape) in zip(scale_factor, input_size)
+    ]
+
+    return output_shape
+
+
+def get_scale_from_size(input_size, output_size):
+    """Get the scale factor given input size and output size.
+
+    Args:
+        input_size (tuple(int)): The size of the input image.
+        output_size (tuple(int)): The size of the output image.
+
+    Returns:
+        list[float]: The scale factor of each dimension.
+    """
+
+    scale = [
+        1.0 * output_shape / input_shape
+        for (input_shape, output_shape) in zip(input_size, output_size)
+    ]
+
+    return scale
+
+
+def cubic(x):
+    """ Cubic function.
+
+    Args:
+        x (ndarray): The distance from the center position.
+
+    Returns:
+        ndarray: The weight corresponding to a particular distance.
+
+    """
+
+    x = np.array(x).astype(np.float32)
+    x_abs = np.absolute(x)
+    x_abs_sq = np.multiply(x_abs, x_abs)
+    x_abs_cu = np.multiply(x_abs_sq, x_abs)
+    f = np.multiply(1.5 * x_abs_cu - 2.5 * x_abs_sq + 1,
```

Comment:
Add a line of note about the math formula used in L61-65

𝑦=1.5𝑥³−2.5𝑥²+1 if |𝑥| <= 1, ...